gems/sorbet-runtime/lib/types/props/serializable.rb

```diff
@@ -261,17 +261,32 @@ def add_prop_definition(prop, rules)
   end
 
   def message_with_generated_source_context(error, generated_method, generate_source_method)
-    line_label = error.backtrace.find {|l| l.end_with?("in `#{generated_method}'")}
-    return unless line_label
+    generated_method = generated_method.to_s
+    if error.backtrace_locations
+      line_loc = error.backtrace_locations.find {|l| l.base_label == generated_method}
+      return unless line_loc
 
-    line_num = if line_label.start_with?("(eval)")
-      # (eval):13:in `__t_props_generated_serialize'
-      line_label.split(':')[1]&.to_i
+      line_num = line_loc.lineno
     else
-      # (eval at /Users/jez/stripe/sorbet/gems/sorbet-runtime/lib/types/props/has_lazily_specialized_methods.rb:65):13:in `__t_props_generated_serialize'
-      line_label.split(':')[2]&.to_i
+      label = if RUBY_VERSION >= "3.4"
+        # in 'ClassName#__t_props_generated_serialize'"
+        "##{generated_method}'"
+      else
+        # in `__t_props_generated_serialize'"
+        "in `#{generated_method}'"
+      end
+      line_label = error.backtrace.find {|l| l.end_with?(label)}
+      return unless line_label
+
+      line_num = if line_label.start_with?("(eval)")
+        # (eval):13:in ...
+        line_label.split(':')[1]&.to_i
+      else
+        # (eval at /Users/jez/stripe/sorbet/gems/sorbet-runtime/lib/types/props/has_lazily_specialized_methods.rb:65):13:in ...
+        line_label.split(':')[2]&.to_i
+      end
+      return unless line_num
```

Comment:
I think you want to keep this `return` outside of the `if`; the original PR for 3.4 compat returned early:

https://github.com/sorbet/sorbet/pull/7710/files#diff-b6758eb606992ab00fd8e66aa0375c569648250f874bf6fb51fe402769e0e9cbL264-L274

unless we are guaranteed to have a `line_num` in the `backtrace_locations` case?

Reply:
> unless we are guaranteed to have a `line_num` in the `backtrace_locations` case?

It's guaranteed if the method is implemented in Ruby, but could be `0` if implemented in C.

But it's a good catch, I applied the suggested change.